cirq/sim/density_matrix_simulator.py

```diff
@@ -214,16 +214,15 @@ def _run_sweep_repeat(
         circuit: circuits.Circuit,
         repetitions: int,
         qubit_order: ops.QubitOrderOrList,
-        intermediate_state: np.ndarray,
+        acton_args: act_on_density_matrix_args.ActOnDensityMatrixArgs,
     ) -> Dict[str, np.ndarray]:
         measurements = {}  # type: Dict[str, List[np.ndarray]]
 
         for _ in range(repetitions):
-            all_step_results = self._base_iterator(
+            all_step_results = self._core_iterator(
                 circuit,
                 qubit_order=qubit_order,
-                initial_state=intermediate_state,
-                is_raw_state=True,
+                sim_state=acton_args.copy(),
```

Comment:
Yes, in the old _base_iterator, you can see it creates a copy.

```
    if np.may_share_memory(initial_matrix, initial_state):
        initial_matrix = initial_matrix.copy()
```

With this PR, `_run_sweep_sample` no longer calls `_base_iterator` and calls `_core_iterator` instead. If anything, this approach should be faster, as it's a straight copy instead of doing `qis.to_valid_density_matrix` and then copying.